financialaid/views.py

```diff
@@ -1,24 +1,205 @@
     authentication_classes = (SessionAuthentication, )
     permission_classes = (IsAuthenticated, )
 
-    def get_queryset(self):
+    def get_queryset(self):  # pragma: no cover
         """
-        Used for returning the view, which hasn't been defined yet.
+        Allows the DRF helper pages to load - not available in production
         """
         return None
+
+
+class ReviewFinancialAidView(UserPassesTestMixin, ListView):
+    """
+    View for reviewing financial aid requests.
+    Note: In the future, it may be worth factoring out the code for sorting into its own subclass of ListView
+    """
+    paginate_by = 10
+    context_object_name = "financial_aid_objects"
+    template_name = "review_financial_aid.html"
+    # If user doesn't pass test_func, raises exception instead of redirecting to login url
+    raise_exception = True
+    # Used to modify queryset and in context
+    selected_status = None
+    program = None
+    # Used for sorting
+    sort_field = None
+    sort_direction = ""
+    sort_field_info = None
+    sort_fields = {
+        "first_name": {
+            "display": "First Name"
+        },
+        "last_name": {
+            "display": "Last Name"
+        },
+        "location": {
+            "display": "Location"
+        },
+        "adjusted_cost": {
+            "display": "Adjusted Cost"
+        },
+        "reported_income": {
+            "display": "Reported Income"
+        },
+    }
+    sort_field_mappings = {
+        "first_name": "user__profile__first_name",
+        "last_name": "user__profile__last_name",
+        "location": "user__profile__city",
+        "adjusted_cost": "adjusted_price",
+        "reported_income": "income_usd",
+    }
+    default_sort_field = "first_name"
+
+    def test_func(self):
+        """
+        Validate user permissions (Analogous to permissions_classes for DRF)
+        """
+        self.program = get_object_or_404(Program, id=self.kwargs.get("program_id", None))
+        return has_object_permission(Permissions.CAN_EDIT_FINANCIAL_AID, self.request.user, self.program)
+
+    def get_context_data(self, **kwargs):
+        """
+        Gets context for view
+        """
+        context = super().get_context_data(**kwargs)
+
+        # Constants required in view
+        context["selected_status"] = self.selected_status
+        context["current_program_id"] = self.program.id
+        # Note: This implementation of retrieving a course price is a naive lookup that assumes
+        # all course runs and courses will be the same price for the foreseeable future,
+        # irrespective of the program. Therefore we can just take the price from any currently
+        # enroll-able course run.
+        course_price = CoursePrice.objects.filter(
+            is_valid=True,
+            course_run__in=CourseRun.objects.filter(CourseRun.get_active_enrollment_queryset())
+        ).first()
+        context["course_price"] = course_price.price
```

Comment:
This would be an issue with the initial configuration of the course/course runs.  If there's a particular exception you think we should raise, we can definitely add it in - we're just not familiar enough with how the initial set up process works to know what is a meaningful exception (unless you recommend creating a custom exception or just raising a generic exception with a helpful message).